collections/nemo_asr/nemo_asr/audio_preprocessing.py

```diff
@@ -0,0 +1,470 @@
+            False, uses torch.stft.
+            Defaults to False
+    """
+    def __init__(
+            self, *,
+            sample_rate=16000,
+            window_size=0.02,
+            window_stride=0.01,
+            window="hann",
+            normalize="per_feature",
+            n_fft=None,
+            preemph=0.97,
+            features=64,
+            lowfreq=0,
+            highfreq=None,
+            log=True,
+            dither=1e-5,
+            pad_to=16,
+            frame_splicing=1,
+            stft_conv=False,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.featurizer = FilterbankFeatures(
+            sample_rate=sample_rate,
+            window_size=window_size,
+            window_stride=window_stride,
+            window=window,
+            normalize=normalize,
+            n_fft=n_fft,
+            preemph=preemph,
+            nfilt=features,
+            lowfreq=lowfreq,
+            highfreq=highfreq,
+            log=log,
+            dither=dither,
+            pad_to=pad_to,
+            frame_splicing=frame_splicing,
+            stft_conv=stft_conv,
+            logger=self._logger
+        )
+        self.featurizer.to(self._device)
+
+    def get_features(self, input_signal, length):
+        return self.featurizer(input_signal, length)
+
+    def get_seq_len(self, seq_len):
+        return self.featurizer.get_seq_len(seq_len)
+
+
+class AudioToMFCCPreprocessor(AudioPreprocessor):
+    """Preprocessor that converts wavs to MFCCs.
+    Uses torchaudio.transforms.MFCC, and several arguments are the same.
+
+    Args:
+        sample_rate: The sample rate of the audio.
+            Defaults to 16000.
+        window_size: Size of window for fft in seconds. Used to calculate the
+            win_length arg for mel spectrogram, overrides mel_kwargs if set.
+            Defaults to None
+        window_stride: Stride of window for fft in seconds. Used to caculate
+            the hop_length arg for mel spect, overrides mel_kwargs if set.
+            Defaults to None
+        window: Windowing function for fft. can be one of ['hann',
+            'hamming', 'blackman', 'bartlett', None]. Overrides mel_kwargs
+            if set; otherwise, torchaudio defaults to hann.
+            Defaults to None
+        n_mfcc: Number of coefficients to retain
+            Defaults to 64
+        dct_type: Type of discrete cosine transform to use
+        norm: Type of norm to use
+        log: Whether to use log-mel spectrograms instead of db-scaled.
+            Defaults to True.
+        mel_kwargs: Dict of arguments for torchaudio.transforms.MelSpectrogram
+    """
+    def __init__(
+            self, *,
+            sample_rate=16000,
+            window_size=None,
+            window_stride=None,
+            window=None,
+            n_mfcc=64,
+            dct_type=2,
+            norm='ortho',
+            log=True,
+            mel_kwargs=None,
```

Comment:
Please don't pass dictionaries to functions. If we cannot express the functionality as a parameter, I would prefer we just don't have that functionality rather than allow it to be passed as a dictionary.

Reply:
torchaudio uses a dictionary argument here.
So do we prefer to "flatten" it in the constructor and then create the dict within this class to pass in? Or not support it? I'd assume the former...

Reply:
Done--expanded args into the AudioToMFCCPreprocessor.